L1Trigger/TrackFindingTracklet/src/PurgeDuplicate.cc

```diff
@@ -316,7 +308,7 @@ void PurgeDuplicate::execute(std::vector<Track>& outputtracks, unsigned int iSec
         for (unsigned int itrk = 0; itrk < numStublists - 1; itrk++) {
           for (unsigned int jtrk = itrk + 1; jtrk < numStublists; jtrk++) {
             // Merge a track with its first duplicate found.
```

Comment:
here is said merge with the first duplicate found but I no not see any break, so what prevents to merge with all others?)

If really one merges with just the first found than there is no need of the dupMap: it is enough for each track to keep record of the first duplicate...

I think the whole algorithm need a review as most probably all those large maps and vectors are not needed...